NetSDK/Services/Client/Classes/SplitClient.cs

```diff
@@ -255,19 +223,94 @@ protected TreatmentResult GetTreatment(Key key, ParsedSplit split, Dictionary<st
             }
         }
 
-        protected virtual void ImpressionLog<T>(IListener<T> listener, IList<T> impressions)
+        protected virtual void ImpressionLog(List<KeyImpression> impressionsQueue)
         {
-            if (listener != null)
+            if (impressionListener != null)
             {
-                foreach (var imp in impressions)
+                foreach (var imp in impressionsQueue)
                 {
-                    listener.Log(imp);
+                    impressionListener.Log(imp);
                 }
             }
         }
         #endregion
 
         #region Private Methods
+        private Dictionary<string, TreatmentResult> GetTreatmentsResult(Key key, List<string> features, Dictionary<string, object> attributes = null)
+        {
+            var treatmentsForFeatures = new Dictionary<string, TreatmentResult>();
+            var ImpressionsQueue = new List<KeyImpression>();
+
+            CheckClientStatus();
+
+            if (_keyValidator.IsValid(key, nameof(GetTreatments)))
+            {
+                features = _splitNameValidator.SplitNamesAreValid(features, nameof(GetTreatments));
+
+                var start = CurrentTimeHelper.CurrentTimeMillis();
+                var clock = new Stopwatch();
+                clock.Start();
+
+                foreach (var feature in features)
+                {
+                    var treatmentResult = DoGetTreatment(key, feature, attributes, true);
+
+                    treatmentsForFeatures.Add(feature, treatmentResult);
+
+                    ImpressionsQueue.Add(BuildImpression(key.matchingKey, feature, treatmentResult.Treatment, start, treatmentResult.ChangeNumber, LabelsEnabled ? treatmentResult.Label : null, key.bucketingKeyHadValue ? key.bucketingKey : null));
+                }
+
+                if (metricsLog != null)
+                {
+                    metricsLog.Time(SdkGetTreatment, clock.ElapsedMilliseconds);
```

Comment:
Are you logging `sdk.GetTreatment` in metrics for `getTreatment`, `getTreatments`, `getTreatmentWithConfig` and `getTreatmentsWithConfig`. It should be different for each method.

Reply:
You are right. Good catch Mati. Thanks!!